### PR TITLE
chore: autogenerate json schema for dagger.json

### PR DESCRIPTION
**Pull request by jedevc**
Fixes https://github.com/dagger/dagger/issues/8405.

This takes @wingyplus's original work from https://github.com/dagger/dagger/issues/8405#issuecomment-2375861357, and additionally applies my suggested changes from https://github.com/dagger/dagger/issues/8405#issuecomment-2399572899.

I was thinking of this in the context of https://discord.com/channels/707636530424053791/1003718839739105300/1300446424894410836, where it would good to have good-quality docs from the start for a new engine config file format.

### CODE REVIEW

**Comment by jedevc** [comment]
@vikram-dagger where would you like to see this integrated into our docs? Should it just be a sidebar link somewhere, or maybe we could also auto-generate a `.mdx` page that contains just the JSON schema? Not really sure here, I'd mostly just like to have this available somewhere in the docs.

File: cmd/json-schema/main.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		if !slices.Contains(args, target.id) {
+			continue
+		}
+
+		r := new(jsonschema.Reflector)
+		if err := r.AddGoComments("github.com/dagger/dagger", "./core/modules"); err != nil {
+			return err
+		}
+
+		s := r.Reflect(&modules.ModuleConfig{})
```
Comment (wingyplus) [inline cmd/json-schema/main.go:32]: It needs to be:

```suggestion
		if err := r.AddGoComments("github.com/dagger/dagger", target.path); err != nil {
			return err
		}

		s := r.Reflect(target.value)
```

??

Comment (jedevc) [reply cmd/json-schema/main.go]: Bleh, had these changes locally, but somehow missed committing them :sweat_smile:

**Review by wingyplus — APPROVED** [review]
Thank you for pushing this forward. 

LGTM 👍

**Review by shykes — APPROVED** [review]

**Pull request merged**